src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -99,6 +99,7 @@ class ProductController extends FrameworkBundleAdminController
      *
      * URL example: /product/catalog/40/20/id_product/asc
      *
+     * @AdminSecurity("is_granted('create', 'ADMINPRODUCTS_') || is_granted('update', 'ADMINPRODUCTS_') || is_granted('read', 'ADMINPRODUCTS_')")
```

Comment:
As @atomiix suggested on another file you can use `request.get('_legacy_controller')` instead of the hard-coded `'ADMINPRODUCTS_'`

Reply:
Done